borgmatic/commands/completion.py

```diff
@@ -55,3 +65,153 @@ def bash_completion():
             '\ncomplete -o bashdefault -o default -F complete_borgmatic borgmatic',
         )
     )
+
+
+# fish section
+
+
+def has_file_options(action: Action):
+    return action.metavar in (
+        'FILENAME',
+        'PATH',
+    ) or action.dest in ('config_paths',)
+
+
+def has_choice_options(action: Action):
+    return action.choices is not None
+
+
+def has_required_param_options(action: Action):
+    return (
+        action.required is True
+        or action.nargs
+        in (
+            '+',
+            '*',
+        )
+        or '--archive' in action.option_strings
+        or action.metavar in ('PATTERN', 'KEYS', 'N')
```

Comment:
Similar here. This looks like it's matching optional flags, which I'm assuming _is_ the intent because you went to the trouble of picking them out here.. I just don't know why.